deepspeed/runtime/pipe/engine.py

```diff
@@ -638,7 +638,6 @@ def _exec_forward_pass(self, buffer_id):
 
         # Zero out the gradients each time we use the tensor because only the data in
         # tensor changes across batches
-        self._zero_grads(inputs)
```

Comment:
Can you explain why we can delete this? We could also delete the comment before this line.

Reply:
https://github.com/microsoft/DeepSpeed/blob/64c670ef021118ed42a98d5dde249f35878a07a7/deepspeed/runtime/pipe/engine.py#L620-L623  here inputs has become a non-leaf tensor, and its gradient has not been saved. if ```if self.is_pipe_partitioned and not self.is_first_stage()``` is False, not create a new leaf tensor named input,  access inputs' gradient in _zero_grad will trigger a warning.